README.md

```diff
@@ -17,38 +28,97 @@ docker pull ghcr.io/buildwithgrove/path-external-auth-server:latest
 ```
 
 - [PEAS GHCR Package](https://github.com/orgs/buildwithgrove/packages/container/package/path-external-auth-server)
+### Architecture Diagram
+
+```mermaid
+graph TD
+    User[/"<big>PATH<br>User</big>"\]
+    Envoy[<big>Envoy Proxy</big>]
 
-### `GatewayEndpoint` Structure
+    AUTH["PEAS (PATH External Auth Server)"]
+    AUTH_DECISION{Did<br>Authorize<br>Request?}
+    PATH[<big>PATH</big>]
 
-PEAS receives a list of `GatewayEndpoints` that define which endpoints are authorized to use the PATH Gateway.
+    Error[[Error Returned to User]]
+    Result[[Result Returned to User]]
 
-- [`GatewayEndpoint` protobuf definition.](https://github.com/buildwithgrove/path-external-auth-server/blob/main/proto/gateway_endpoint.proto)
+    GroveDB[("Grove Portal Database<br>(Postgres)")]
 
-`GatewayEndpoint` data is received from a `remote gRPC server` that may be implemented by a PATH gateway operator in any way they see fit. The only requirement is that it adhere to the to spec defined in the protobuf definition.
+    subgraph AUTH["PEAS<br/>PATH External Auth Server"]
+    end
+
+    User -->|1.Send Request| Envoy
+    Envoy -.->|2.Authorization Check<br>gRPC| AUTH
+    AUTH -.->|3.Authorization Result<br>gRPC| Envoy
+    Envoy --> AUTH_DECISION
+    AUTH_DECISION -->|4.No <br> Forward Request| Error
+    AUTH_DECISION -->|4.Yes <br> Forward Request| PATH
+    PATH -->|5.Response| Result
+
+    GroveDB <-->|Postgres Connection| AUTH
+```
+
+### `PortalApp` Structure
+
+`PEAS` manages authentication and assigning rate limiting headers for portal apps that are defined with the following structure:
```

Comment:
Can you remove this and just specify where the struct is defined in the code?

I want to maintain one source of truth.

The table below should be enough.

Reply:
Certainly.